src/main/java/listeners/MainEventListener.java

```diff
@@ -275,6 +293,36 @@ private void clear(Member member, TextChannel channel, Server server, String[] a
                 topic.getName())).queue();
     }
 
+    private void finish(Member member, TextChannel channel, Server server, String[] args) {
+        Topic topic = null;
+        Room room = null;
+        
+        // find the topic corresponding to this channel
+        for (Topic t : server.getTopics()) {
+            Optional<Room> optionalRoom = t.getRoom(channel.getName());
+            if (optionalRoom.isPresent()) {
+                topic = t;
+                room = optionalRoom.get();
+            }
+        }
+
+        // only run inside a room
+        if (room == null) {
```

Comment:
Might be cleaner to have `room` be `Optional<Room>` so then you can just check `room.isEmpty()` (or whatever the null check is)